spring-cql/src/test/java/org/springframework/cassandra/test/integration/EmbeddedCassandraServerHelper.java

```diff
@@ -0,0 +1,357 @@
+		String host = DatabaseDescriptor.getRpcAddress().getHostName();
+		int port = DatabaseDescriptor.getNativeTransportPort();
+
+		com.datastax.driver.core.Cluster cluster = com.datastax.driver.core.Cluster.builder().addContactPoint(host)
+				.withPort(port).build();
+		com.datastax.driver.core.Session session = cluster.connect();
+
+		try {
+			List<String> keyspaces = new ArrayList<String>();
+			for (com.datastax.driver.core.KeyspaceMetadata keyspace : cluster.getMetadata().getKeyspaces()) {
+				if (!isSystemKeyspaceName(keyspace.getName())) {
+					keyspaces.add(keyspace.getName());
+				}
+			}
+			for (String keyspace : keyspaces) {
+				session.execute("DROP KEYSPACE " + keyspace);
+			}
+		} finally {
+			session.close();
+			cluster.close();
+
+		}
+	}
+
+	private static boolean isSystemKeyspaceName(String keyspaceName) {
+		return INTERNAL_CASSANDRA_KEYSPACE.equals(keyspaceName) || INTERNAL_CASSANDRA_AUTH_KEYSPACE.equals(keyspaceName)
+				|| INTERNAL_CASSANDRA_TRACES_KEYSPACE.equals(keyspaceName);
+	}
+
+	private static void rmdir(String dir) throws IOException {
+		File dirFile = new File(dir);
+		if (dirFile.exists()) {
+			FileUtils.deleteRecursive(new File(dir));
+		}
+	}
+
+	/**
+	 * Copies a resource from within the jar to a directory.
+	 *
+	 * @param resource
+	 * @param directory
+	 * @throws IOException
+	 */
+	private static void copy(String resource, String directory) throws IOException {
+		mkdir(directory);
+		String fileName = resource.substring(resource.lastIndexOf("/") + 1);
+		File file = new File(directory + System.getProperty("file.separator") + fileName);
+		InputStream is = EmbeddedCassandraServerHelper.class.getResourceAsStream(resource);
+		OutputStream out = new FileOutputStream(file);
+		byte buf[] = new byte[1024];
+		int len;
+		while ((len = is.read(buf)) > 0) {
+			out.write(buf, 0, len);
+		}
+		out.close();
+	}
+
+	/**
+	 * Creates a directory
+	 *
+	 * @param dir
+	 * @throws IOException
+	 */
+	private static void mkdir(String dir) throws IOException {
+		FileUtils.createDirectory(dir);
+	}
+
+	private static void cleanupAndLeaveDirs() throws IOException {
+		mkdirs();
+		cleanup();
+		mkdirs();
+		CommitLog commitLog = CommitLog.instance;
+		commitLog.getContext(); // wait for commit log allocator instantiation to avoid hanging on a race condition
+		commitLog.resetUnsafe(); // cleanup screws w/ CommitLog, this brings it back to safe state
+	}
+
+	private static void cleanup() throws IOException {
+		// clean up commitlog
+		String[] directoryNames = { DatabaseDescriptor.getCommitLogLocation(), };
+		for (String dirName : directoryNames) {
+			File dir = new File(dirName);
+			if (!dir.exists())
+				throw new RuntimeException("No such directory: " + dir.getAbsolutePath());
+			FileUtils.deleteRecursive(dir);
+		}
+
+		// clean up data directory which are stored as data directory/table/data
+		// files
+		for (String dirName : DatabaseDescriptor.getAllDataFileLocations()) {
```

Comment:
Refactor lines 272-278 and lines 282-287 into...

```
void rmdirs(String... directoryNames) {
  for (String directoryName : directoryNames) {
    File directory = new File(directoryName);
    if (!directory.exists()) {
        throw new FileNotFoundException(String.format("No such directory: %s", directory));
    }
    FileUtils.deletedRecursive(directory);
}
```

You could even refactor your `rmdir(:String)` method above (throwing the `RuntimeException` if necessary) and have this **new** `rmdirs(directories:String[])` method call it.